R/ml_problem.R

```diff
@@ -94,27 +112,29 @@ fitting_problem <- function(ref_sample,
                             field_names,
                             individual_controls, group_controls,
                             prior_weights = NULL) {
-  new_fitting_problem(
-    list(
-      refSample = ref_sample, controls = controls, fieldNames = field_names,
-      priorWeights = prior_weights
-    )
+  .Deprecated("ml_problem")
```

Comment:
```suggestion
  deprecate_soft("mlfit::fitting_problem()", "mlfit::ml_problem()")
```

with a suitable import from the lifecycle package.

Reply:
shouldn't this be?

```r
deprecate_soft("0.4.0", "mlfit::fitting_problem()", details = "Please use `mlfit::ml_problem()` instead.")
```

Reply:
Yeah, `deprecate_soft("0.4.0", "mlfit::fitting_problem()", "mlfit::ml_problem()")` should be enough.